### PR TITLE
Added support for cluster roll 

### PR DESCRIPTION
**Pull request by sharadkesarwani**
Added support for cluster roll.

# Jira Ticket
https://spotinst.atlassian.net/browse/SPOTAUT-17448

### CODE REVIEW

File: examples/service/ocean/providers/azure/clusterroll/main.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	out, err := svc.CloudProviderAzure().CreateRoll(ctx, &azure.CreateRollInput{
+		Roll: &azure.RollSpec{
+			ClusterID:                 spotinst.String("o-12345"),
+			BatchSizePercentage:       spotinst.Int(20),
+			Comment:                   spotinst.String("Comment to describe roll."),
+			RespectPDB:                spotinst.Bool(true),
+			BatchMinHealthyPercentage: spotinst.Int(100),
+			NodeNames:                 []string{"node123"},
+			VngIds:                    []string{"vng-123"},
+			NodePoolNames:             []string{"nodepool12345", "nodepool67890"},
```
Comment (chandra1-n) [inline examples/service/ocean/providers/azure/clusterroll/main.go:41]: please add respectRestrictScaleDown in the example

**Review by chandra1-n — CHANGES_REQUESTED** [review]
Please implement support for list rolls and stop roll as well

File: service/ocean/providers/azure/cluster.go [hunk 2]
```diff
@@ -230,6 +230,72 @@ type ManagedServiceIdentity struct {
...
+	Scope                     *string    `json:"scope,omitempty"`
+	Comment                   *string    `json:"comment,omitempty"`
+	Status                    *string    `json:"status,omitempty"`
+	Progress                  *Progress  `json:"progress,omitempty"`
+	RespectPDB                *bool      `json:"respectPdb,omitempty"`
+	RespectRestrictScaleDown  *bool      `json:"respectRestrictScaleDown,omitempty"`
+	BatchMinHealthyPercentage *int       `json:"batchMinHealthyPercentage,omitempty"`
+	CurrentBatch              *int       `json:"currentBatch,omitempty"`
+	NumOfBatches              *int       `json:"numOfBatches,omitempty"`
+	LaunchSpecIDs             []string   `json:"launchSpecIds,omitempty"`
```
Comment (chandra1-n) [inline service/ocean/providers/azure/cluster.go:262]: please re-check with actual response

**Review by chandra1-n — CHANGES_REQUESTED** [review]
please fix the review comments

File: service/ocean/providers/azure/cluster.go (hunk 2 above)
Comment (sharadkesarwani) [reply service/ocean/providers/azure/cluster.go:262]: Removed LaunchSpecIDs and InstanceIDs attributes from structs as those were mistakenly added.

**Comment by sharadkesarwani** [comment]
> Please implement support for list rolls and stop roll as well

Added support for list rolls and stop roll

**Comment by sharadkesarwani** [comment]
> please fix the review comments

Fixed all review comments.

**Review by chandra1-n — APPROVED** [review]

**Pull request merged**